### ISSUE TITLE
Redirect loop on SAML authentication 

### ISSUE THREAD

**Issue by simao-silva** [salesagility/SuiteCRM#9888]
#### Issue

I am trying to set up SAML authentication. I have enable SAML authentication in Password Management and fill in the boxes as followed:

- Login URL: `https://<KEYCLOAK URL>/auth/realms/master/protocol/saml/clients/suitecrm`
- SLO URL: `https://<KEYCLOAK URL>/auth/realms/master/protocol/saml/`
- X509 Certificate: `MIICm.....`

The login is redirected as expected to Keycloak. However, after I fill out the username and password, a redirect loop starts with 2 requests:
- A GET request with status code 200 to `https://<KEYCLOAK URL>/auth/realms/master/protocol/saml/clients/suitecrm?SAMLRequest=***************&RelayState=https%3A%2F%2F******%2Findex.php`
- A GET request with status code 302 to `https://<SuiteCRM URL>/index.php?action=Login&module=Users` that in the *Location* header shows my Keycloak URL. 

#### Expected Behavior

Show SuiteCRM website after login with user authenticated. 

#### Actual Behavior

Continuous redirection loop after successful login.

#### Steps to Reproduce

1. Configure SAML authentication in `Password Management`
2. Open a browser and go to SuiteCRM URL and expect the redirection to SAML provider
3. Login

#### Your Environment

* SuiteCRM Version used: 7.12.4
* Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)): Chrome 108.0.5359.124 and Firefox 108.0.1
* Environment name and version (e.g. MySQL, PHP 7): [Bitnami docker image](https://hub.docker.com/r/bitnami/suitecrm) with MariaDB 10.6
* Operating System and version (e.g Ubuntu 16.04): Debian 10

**Comment by vladaman** [comment]
It worked well with JumpCloud. We tried to migrate to Keycloak and we experienced the same issue.

**Url stream:**

1. KeyCloak /realms/[realm]/protocol/saml
2. crm.site.com/saml/acs
3. crm.site.com/logged-out
4. crm.site.com/index.php
5. crm.site.com/index.php/saml/login
6. KeyCloak again... (_loop continues_)

We use SuiteCRM 8.7.1 and

**Comment by vladaman** [comment]
Yes! We’ve got it working too. It’s not perfect—logout still needs work—but getting login to work is a big win.

```env
# AUTH_TYPE=native
###> SAML CONFIG ###
SAML_USERNAME_ATTRIBUTE=email
SAML_USE_ATTRIBUTE_FRIENDLY_NAME=false
SAML_AUTO_CREATE=enabled
###< SAML CONFIG ###

SAML_IDP_ENTITY_ID='https://auth.site.com/realms/site'
SAML_IDP_SSO_URL='https://auth.site.com/realms/site/protocol/saml'
SAML_IDP_SLO_URL='https://auth.site.com/realms/site/protocol/saml'
SAML_IDP_X509CERT='MII.........'
SAML_SP_ENTITY_ID=''
SAML_SP_PRIVATE_KEY=''
SAML_SP_CERT=''
SAML_STRICT='false'
SAML_DEBUG='true'
```

<img width="420" alt="Keycloak Screenshot" src="https://github.com/user-attachments/assets/94cc3017-8f15-4692-a512-60ca7b4fc6c4" />
